app/scripts/controllers/network/network-controller.js

```diff
@@ -447,13 +452,30 @@ export default class NetworkController extends EventEmitter {
   }
 
   _setNetworkClient({ networkMiddleware, blockTracker }) {
+    const networkProvider = providerFromMiddleware(networkMiddleware);
```

Comment:
@Gudahtt Is this what you were looking for? After this I am not sure if I would be able to tell the difference between the "MetaMask middleware" and the "network middleware", but I guess this feels better...

Reply:
Yeah this is exactly what I had in mind!

The "metamask middleware" is being removed in this PR: https://github.com/MetaMask/metamask-extension/pull/16863 . It won't be here long.